queasars/minimum_eigensolvers/evqe/evolutionary_algorithm/selection.py

```diff
@@ -28,13 +29,36 @@ class EVQESelection(BaseEvolutionaryOperator[EVQEPopulation]):
     :type: float
     :param beta_penalty: scaling factor for penalizing the amount of controlled gates of an individual
     :type: float
+    :param use_tournament_selection: indicates whether to use tournament selection. By default, this is
+        set to False. In that case, roulette wheel selection is used. Should be true, if the measured expectation
+        values can be negative.
+    :type use_tournament_selection: bool
+    :param tournament_size: indicates the size of the tournaments used. This can be in the range [1, population_size].
+        It cannot be None, if use_tournament_selection is set to True. A tournament_size of 1 yields random selection,
+        with increasing tournament selection sizes increasing the selection pressure.
+    :type tournament_size: int
     :param random_seed: integer value to control randomness
     :type: int
     """
 
-    def __init__(self, alpha_penalty: float, beta_penalty: float, random_seed: Optional[int]):
+    def __init__(
+        self,
+        alpha_penalty: float,
+        beta_penalty: float,
+        use_tournament_selection: bool = False,
+        tournament_size: Optional[int] = None,
+        random_seed: Optional[int] = None,
+    ):
         self.alpha_penalty: float = alpha_penalty
         self.beta_penalty: float = beta_penalty
+        self.use_tournament_selection: bool = use_tournament_selection
```

Comment:
Aren't those fields internal? I mean the selection operator is in particular applied via `apply_operator`, so all these fields are rather internal implementation details, no?

Reply:
You are correct. I changed these variables to be private.